### PR TITLE
Arraylist<LayoutBlock> GSON anonymous inner class initialization bug

### PR DESCRIPTION
**Pull request by WilliamBergamin**
GSON has a [known issue](https://github.com/google/gson/issues/2023#issuecomment-974230703) where null is returned if it tries to serialize an object that was initialized using an anonymous inner class.

Example:
```java
import com.google.gson.Gson;
import com.google.gson.GsonBuilder;

import java.util.ArrayList;
import java.util.List;

List<String> strings = new ArrayList<>() {{
            add("hello");
            add("world");
        }};

GsonBuilder builder = new GsonBuilder();
Gson gson = builder.create();

String json = gson.toJson(strings);

assertEquals("{\"hello\", \"world\"}", json); // this should be true but json = "null"
```

The above assertion should be true but the json string generated by the GSON lib actually returns a string of value "null". This can lead to some head scratching for the users of this library.

These changes try to reinitialize & reserialize the list of blocks if this "null" output is obtained from the GSON library.

### Category (place an `x` in each of the `[ ]`)

* [ ] **bolt** (Bolt for Java)
* [ ] **bolt-{sub modules}** (Bolt for Java - optional modules)
* [x] **slack-api-client** (Slack API Clients)
* [ ] **slack-api-model** (Slack API Data Models)
* [ ] **slack-api-*-kotlin-extension** (Kotlin Extensions for Slack API Clients)
* [ ] **slack-app-backend** (The primitive layer of Bolt for Java)

## Requirements

Please read the [Contributing guidelines](https://github.com/slackapi/java-slack-sdk/blob/main/.github/contributing.md) and [Code of Conduct](https://slackhq.github.io/code-of-conduct) before creating this issue or pull request. By submitting, you are agreeing to the those rules.

### CODE REVIEW

File: slack-api-client/src/main/java/com/slack/api/methods/RequestFormBuilder.java
Comment (filmaj) [inline slack-api-client/src/main/java/com/slack/api/methods/RequestFormBuilder.java:2559]: I'd add a comment here that explains _why_ we need this method, just as a hint to a maintainer reading through this code in the future (I have a bad memory so I probably won't remember the details of this problem in 6 months!). Maybe just a quick link to the GSON issue and a single sentence explanation.

**Review by filmaj — APPROVED** [review]
LGTM, but will defer to @seratch for approval

File: slack-api-client/src/test/java/test_locally/api/methods/RequestFormBuilderTest.java
Comment (seratch) [inline slack-api-client/src/test/java/test_locally/api/methods/RequestFormBuilderTest.java:34]: I like the test structure. If you want to have these comments, probably Given-When-Then style would be more common.

File: slack-api-client/src/main/java/com/slack/api/methods/RequestFormBuilder.java
Comment (seratch) [reply slack-api-client/src/main/java/com/slack/api/methods/RequestFormBuilder.java:2559]: Sounds good. Adding a link to the gson project issue would be easier.

File: slack-api-client/src/test/java/test_locally/api/methods/RequestFormBuilderTest.java
Comment (seratch) [inline slack-api-client/src/test/java/test_locally/api/methods/RequestFormBuilderTest.java:35]: You can test one of the public methods that internally calls getJsonWithGsonAnonymInnerClassHandling instead of trying to run the private method using reflection APIs.

File: slack-api-client/src/main/java/com/slack/api/methods/RequestFormBuilder.java
Comment (seratch) [inline slack-api-client/src/main/java/com/slack/api/methods/RequestFormBuilder.java:1138]: `req.getUsers(): List<String>` patterns also need the same workaround.

**Review by seratch — CHANGES_REQUESTED** [review]

File: slack-api-client/src/main/java/com/slack/api/methods/RequestFormBuilder.java
Comment (WilliamBergamin) [reply slack-api-client/src/main/java/com/slack/api/methods/RequestFormBuilder.java]: Agreed, I added a comment with the link to the original issue

File: slack-api-client/src/main/java/com/slack/api/methods/RequestFormBuilder.java
Comment (WilliamBergamin) [reply slack-api-client/src/main/java/com/slack/api/methods/RequestFormBuilder.java:1136]: Makes sense, I also noticed that the Map objects also suffered from this GSON issue, I added additional code to handle these use cases.

**Review by seratch — APPROVED** [review]
Great improvement 💯  Thanks for fixing this!

**Pull request merged**
